packages/flags/src/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { Flags, Flag } from './index';
+
+const FF = 'ASAP_FF';
+
+export const isValidFF = (str: string, flags: Flags): boolean => {
+  try {
+    const parsedValue = JSON.parse(str);
+
+    if (Array.isArray(parsedValue)) return false;
+
+    if (Object.keys(parsedValue).length === 0) return false;
+
+    return Object.entries(parsedValue).some(
+      ([key, val]) =>
+        Object.keys(flags).includes(key) && typeof val === 'boolean',
+    );
+  } catch (e) {
+    return false;
+  }
+};
+export const isFF = (name: string, value: string, flags: Flags): boolean =>
+  name.trim() === FF && isValidFF(value, flags);
+
+export const mergeFlags = (flags: Flags): Partial<Record<Flag, boolean>> => {
+  if (process.browser) {
+    return document.cookie
```

Comment:
It assumes having a single cookie that stores a JSON value `{ [feature_flag]: flag }`
Instead of setting a cookie for each feature flag.